src/masoniteorm/query/QueryBuilder.py

```diff
@@ -1546,6 +1546,10 @@ def when(self, conditional, callback):
             callback(self)
         return self
 
+    def truncate(self):
+        self.get_grammar().truncate_table(self._table)
```

Comment:
You might have to do something like

```python
def truncate(self):
     sql = self.get_grammar().truncate_table(self._table)
     if self._dry:
        return sql

     self.new_connection().query(sql, ())
```

Reply:
@girardinsamuel That's the solution. When the querybuilder are in "dry" mode it can return just the sql generated, instead it hit the database.

Reply:
- I am a bit surprised that this kind of logic is not present for other type of queries (drop, rename, ...) ?
- Also does this assume that query builders tests are running with dry mode enabled ?

Reply:
1. In Schema.py, the methods: truncate, drop_if_exists and truncate there the verification of the dry mode or to hit database, just the `drop` don't.
2. Once the Schema or QueryBuilder is built, there a flag to indicate that in constructor of them in tests I think.

Reply:
> In Schema.py, the methods: truncate, drop_if_exists and truncate there the verification of the dry mode or to hit database, just the drop don't.

that's correct here. The code is built to be tested first so there are flags we pass to make things dry so we don't need a database just to test queries. If the queries we assert are correct then the actual database calls will 100% work. So no need to make complexity by testing actual database connections.

> Once the Schema or QueryBuilder is built, there a flag to indicate that in constructor of them in tests I think.

also correct. The tests pass dry=True in the test case classes query builder initializer. Most test case classes have a get_builder() method where we set the dry parameter to `True`. Some might not but that's because those ones are just building a query and asserting the SQL. 

There are also some methods like I think create and update that take a query=True parameter that does the same thing but this is largely an inconsistency in the codebase